src/drivers/nuopc/cime_flds/esmFldsExchange.F90

```diff
@@ -1109,62 +1112,79 @@ subroutine esmFldsExchange(gcomp, phase, rc)
-          ! 'fractional atmosphere coverage used in radiation computations wrt ocean'
+          ! 'fractional atmosphere coverage used in radiation computations wrt ocean' (computed in med_phases_prep_ocn)
           call addfld(fldListTo(compocn)%flds, 'Sf_afracr')
-       end if
-    else
-       if (flds_i2o_per_cat) then
-          call addmap(fldListFr(compice)%flds, 'Si_ifrac_n', compocn, mapfcopy, 'unset', 'unset')
+          ! 'net shortwave radiation times atmosphere fraction' (computed in med_phases_prep_ocn)
+          call addfld(fldListTo(compocn)%flds, 'Foxx_swnet_afracr')
+       else
+          call addmap(fldListFr(compice)%flds, 'Si_ifrac_n'        , compocn, mapfcopy, 'unset', 'unset')
           call addmap(fldListFr(compice)%flds, 'Fioi_swpen_ifrac_n', compocn, mapfcopy, 'unset', 'unset')
-          ! TODO (mvertens, 2018-12-21): add mapping and merging
+          ! Note that 'Sf_afrac, 'Sf_afracr' and 'Foxx_swnet_afracr' will have explicit merging in med_phases_prep_ocn
        end if
     end if
 
     ! ---------------------------------------------------------------------
     !  to ocn: precipitation rate water equivalent from atm
     !  to ocn: snow rate water equivalent from atm
     ! ---------------------------------------------------------------------
+
     if (phase == 'advertise') then
-       do n = 1,size(iso)
-          call addfld(fldListFr(compatm)%flds, 'Faxa_rainc'//iso(n))
-          call addfld(fldListFr(compatm)%flds, 'Faxa_rainl'//iso(n))
-          call addfld(fldListFr(compatm)%flds, 'Faxa_rain' //iso(n))
-          call addfld(fldListTo(compocn)%flds, 'Faxa_rain' //iso(n))
-
-          call addfld(fldListFr(compatm)%flds, 'Faxa_snowc'//iso(n))
-          call addfld(fldListFr(compatm)%flds, 'Faxa_snowl'//iso(n))
-          call addfld(fldListFr(compatm)%flds, 'Faxa_snow' //iso(n))
-          call addfld(fldListTo(compocn)%flds, 'Faxa_snow' //iso(n))
-       end do
+       call addfld(fldListFr(compatm)%flds, 'Faxa_rainc')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_rainl')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_rain' )
+       call addfld(fldListTo(compocn)%flds, 'Faxa_rain' )
+
+       call addfld(fldListFr(compatm)%flds, 'Faxa_rainc_wiso')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_rainl_wiso')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_rain_wiso' )
+       call addfld(fldListTo(compocn)%flds, 'Faxa_rain_wiso' )
+
+       call addfld(fldListFr(compatm)%flds, 'Faxa_snowc')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_snowl')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_snow' )
+       call addfld(fldListTo(compocn)%flds, 'Faxa_snow' )
+
+       call addfld(fldListFr(compatm)%flds, 'Faxa_snowc_wiso')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_snowl_wiso')
+       call addfld(fldListFr(compatm)%flds, 'Faxa_snow_wiso' )
+       call addfld(fldListTo(compocn)%flds, 'Faxa_snow_wiso' )
     else
-       do n = 1,size(iso)
+       do n = 1,2
```

Comment:
seems like you are removing some flexibility and replacing it with hard coded values.